accessors/storage/storage_accessor.go

```diff
@@ -0,0 +1,192 @@
+						// you would use a condition such as "matchesPrefix":["pictures/paris_"].
+						MatchesPrefix: matchesPrefix,
+					},
+				},
+			},
+		}
+	}
+
+	if err := bucket.Create(ctx, projectID, &attrs); err != nil {
+		if e, ok := err.(*googleapi.Error); ok {
+			// Ignoring the bucket already exists error.
+			if e.Code != 409 {
+				return fmt.Errorf("failed to create bucket: %v", err)
+			} else {
+				fmt.Printf("Using the existing bucket: %v \n", bucketName)
+			}
+		} else {
+			return fmt.Errorf("failed to create bucket: %v", err)
+		}
+
+	} else {
+		fmt.Printf("Created new GCS bucket: %v\n", bucketName)
+	}
+	return nil
+}
+
+// Applies the bucket lifecycle with delete rule. Only accepts the Age and
+// prefix rule conditions as it is only used for the Datastream destination
+// bucket currently.
+func EnableBucketLifecycleDeleteRule(ctx context.Context, bucketName string, matchesPrefix []string, ttl int64) error {
+	client, err := storageclient.GetOrCreateClient(ctx)
+	if err != nil {
+		return fmt.Errorf("could not create client while enabling lifecycle: %w", err)
+	}
+
+	for i, str := range matchesPrefix {
+		matchesPrefix[i] = strings.TrimPrefix(str, "/")
+	}
+	bucket := client.Bucket(bucketName)
+	bucketAttrsToUpdate := storage.BucketAttrsToUpdate{
+		Lifecycle: &storage.Lifecycle{
+			Rules: []storage.LifecycleRule{
+				{
+					Action: storage.LifecycleAction{Type: "Delete"},
+					Condition: storage.LifecycleCondition{
+						AgeInDays: ttl,
+						// The prefixes should not contain the bucket names and starting slash.
+						// For object gs://my_bucket/pictures/paris_2022.jpg,
+						// you would use a condition such as "matchesPrefix":["pictures/paris_"].
+						MatchesPrefix: matchesPrefix,
+					},
+				},
+			},
+		},
+	}
+
+	attrs, err := bucket.Update(ctx, bucketAttrsToUpdate)
+	if err != nil {
+		return fmt.Errorf("could not bucket with lifecycle: %w", err)
+	}
+	logger.Log.Info(fmt.Sprintf("Added lifecycle rule to bucket %v\n. Rule Action: %v\t Rule Condition: %v\n",
+		bucketName, attrs.Lifecycle.Rules[0].Action, attrs.Lifecycle.Rules[0].Condition))
+	return nil
+}
+
+// UploadLocalFileToGCS uploads an object.
+func UploadLocalFileToGCS(ctx context.Context, filePath, fileName, localFilePath string) error {
+	data, err := os.ReadFile(localFilePath)
+	if err != nil {
+		return fmt.Errorf("could not read file %s: %w", localFilePath, err)
+	}
+	return WriteDataToGCS(ctx, filePath, fileName, string(data))
+}
+
+func WriteDataToGCS(ctx context.Context, filePath, fileName, data string) error {
+	client, err := storageclient.GetOrCreateClient(ctx)
+	if err != nil {
+		return fmt.Errorf("could not create client while uploading to GCS: %w", err)
+	}
+
+	u, err := utils.ParseGCSFilePath(filePath)
+	if err != nil {
+		return fmt.Errorf("parseFilePath: unable to parse file path: %v", err)
+	}
+	bucketName := u.Host
+	bucket := client.Bucket(bucketName)
+	obj := bucket.Object(u.Path[1:] + fileName)
+
+	w := obj.NewWriter(ctx)
```

Comment:
It will be good to print time taken to load file to GCS.

Reply:
Added log statements here, the logger prints time stamp as well.